nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb

```diff
@@ -290,29 +291,62 @@ def dependency_has_search_results?(dependency)
             dependency_urls = [UpdateChecker::RepositoryFinder.get_default_repository_details(dependency.name)]
           end
           dependency_urls_with_package = dependency_urls.select do |dependency_url|
-            response = execute_search_for_dependency_url(dependency_url)
-            next unless response.status == 200
+            dependency_url_has_matching_result?(dependency.name, dependency_url)
+          end
 
-            body = JSON.parse(response.body)
-            data = body["data"]
-            next unless data.length.positive?
+          dependency_urls_with_package.any?
+        end
 
-            found_matching_result = data.any? { |result| result["id"].casecmp?(dependency.name) }
-            found_matching_result
+        def dependency_url_has_matching_result?(dependency_name, dependency_url)
+          repository_type = dependency_url.fetch(:repository_type)
+          if repository_type == "v3"
+            dependency_url_has_matching_result_v3?(dependency_name, dependency_url)
+          elsif repository_type == "v2"
+            dependency_url_has_matching_result_v2?(dependency_name, dependency_url)
+          else
+            raise "Unknown repository type: #{repository_type}"
           end
+        end
 
-          dependency_urls_with_package.any?
+        def dependency_url_has_matching_result_v3?(dependency_name, dependency_url)
+          url = dependency_url.fetch(:search_url)
+          auth_header = dependency_url.fetch(:auth_header)
+          response = execute_search_for_dependency_url(url, auth_header)
+          return false unless response.status == 200
+
+          body = JSON.parse(response.body)
+          data = body["data"]
+          return false unless data.length.positive?
+
+          found_matching_result = data.any? { |result| result["id"].casecmp?(dependency_name) }
+          found_matching_result
```

Comment:
Could we do the following instead?

```suggestion
          data.any? { |result| result["id"].casecmp?(dependency_name) }
```

Reply:
Done.